scripts/gulpfile.js

```diff
@@ -66,15 +108,77 @@ gulp.task('stories/26px', () => {
   return file('26px.js', indexJS, { src: true }).pipe(gulp.dest('src/.stories'));
 });
 
-gulp.task('stories', gulp.parallel('stories/12px', 'stories/16px', 'stories/26px'));
+gulp.task('stories/components', () => {
+  const indexJS = toIndexFile('dist/components');
+
+  return file('components.js', indexJS, { src: true }).pipe(gulp.dest('src/.stories'));
+});
+
+gulp.task(
+  'stories',
+  gulp.parallel('stories/12px', 'stories/16px', 'stories/26px', 'stories/components')
+);
 
 const source = path.join('src', '**', '*.svg');
 
-gulp.task('dist', () => {
+gulp.task('dist/sprite', () => {
   return gulp.src(source).pipe(svgSprite(config)).pipe(gulp.dest('dist'));
 });
 
-gulp.task('default', gulp.parallel('dist', 'stories'));
+const transformToComponent = svgCode =>
+  `/**
+* Copyright Zendesk, Inc.
+*
+* Use of this source code is governed under the Apache License, Version 2.0
+* found at http://www.apache.org/licenses/LICENSE-2.0.
+*/
+`.concat(
+    transform.sync(
+      svgCode,
+      {
+        plugins: ['@svgr/plugin-svgo', '@svgr/plugin-jsx', '@svgr/plugin-prettier'],
+        index: false,
+        svgoConfig,
+        prettierConfig: './.prettierrc'
+      },
+      { componentName: 'ReactIcon' }
+    )
+  );
+
+const toCamelCase = text => {
+  return text
+    .replace(/(?:^\w|[A-Z]|\b\w)/gu, leftTrim => leftTrim.toUpperCase())
+    .replace(/\s+/gu, '')
+    .replace(/-/gu, '');
+};
+
+gulp.task('dist/components', () => {
+  return gulp
+    .src(source)
+    .pipe(
+      rename(filePath => ({
+        ...filePath,
+        basename: toCamelCase(filePath.basename),
+        extname: '.js'
+      }))
+    )
+    .pipe(
+      through2.obj((svgFile, _, cb) => {
+        if (svgFile.isBuffer()) {
```

Comment:
Are there cases when the SVG file is not loaded as a buffer?

Reply:
I used the [example from the Gulp doc](https://gulpjs.com/docs/en/getting-started/using-plugins#inline-plugins) as example. I think it's good to leave it as a safety net to avoid the generation failing if there's a non buffer file going through